File: BestTimeToBuyAndSellStock.py
class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        if len(prices) <= 1: return 0
        low = prices[0]
        maxprofit = 0
        for i in range(len(prices)):
            if prices[i] < low: low = prices[i]
            maxprofit = max(maxprofit, prices[i] - low)
        return maxprofit

File: CPP/Second/best_time_to_buy_and_sell_stock.cc
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        if (prices.size() <= 1)
            return 0;
        int low = prices[0], res = 0;
        for (int i = 1; i < prices.size(); i++) {
            res = max(res, prices[i] - low);
            low = min(low, prices[i]);
        }
        return res;
    }
};

int main(int argc, char *argv[])
{

    return 0;
}
